prow/pubsub/subscriber/subscriber.go

```diff
@@ -199,6 +205,12 @@ func (prh *presubmitJobHandler) getProwJobSpec(cfg prowCfgClient, pc *config.InR
 	var presubmitJob *config.Presubmit
 	org, repo, branch := refs.Org, refs.Repo, refs.BaseRef
 	orgRepo := org + "/" + repo
+	// Add "https://" prefix to orgRepo if this is a gerrit job.
+	// (Unfortunately gerrit jobs use the full repo URL as the identifier.)
+	prefix := "https://"
+	if pe.Labels[client.GerritRevision] != "" && !strings.HasPrefix(orgRepo, prefix) {
```

Comment:
while I like the idea to explicitly check whether it's a gerrit job or not, are we sure that this label is reliable enough? In another word, if a user forgot to set this label in their job, the response they got would be something like `failed to locate prow job`, instead of `gerrit-revision label must be provided`.

How about try looking up prow jobs with `<orgRepo>` first, if not found then try `https://<orgRepo>`?

Reply:
> are we sure that this label is reliable enough?

I think it is, it is not a label that users would configure on a job, this is set by the gerrit adapter or in the pubsub trigger message. Also reporting logic seems to use this as a differentiation already: https://github.com/kubernetes/test-infra/blob/e267164240be4d7b4d11cfe9258668687514a48c/prow/crier/reporters/gerrit/reporter.go#L189-L195

I was originally going to check both like you are suggesting, but I didn't want gerrit jobs to be able to be configured without the `https://` prefix or GitHub jobs to be configured with it. Its not great that gerrit jobs currently use the wrong format, but at least we have consistent formatting per platform right now. If we allow either format for either platform existing assumptions may be broken.

Reply:
I see. In this case I think we would make it clear in our documentations that if this is a gerrit job, it's required to set this label, for example at https://github.com/kubernetes/test-infra/tree/master/prow/cmd/sub#presubmit-prow-jobs